10up/autoshare-for-twitter
# Migrate to Twitter API v2

### Description of the Change
PR migrates the current Twitter API implementation to use API v2. Majorly PR has made the below changes
- Migrated to Twitter API v2
- Add migrate to v2 notice for only existing users
- Update app setup steps to the latest process
- Added learn more link to error message for old credentials user


Closes #173

### How to test the Change
Follow the steps mentioned on the `Autoshare for Twitter Settings`  admin page and set up the Twitter app. You should be able to set up the plugin without struggle. After saving the plugin setting with the Twitter app details, you should be able to tweet from WordPress from the post edit page. 

1. Make sure the plugin works fine with Twitter API v2
2. Verify plugin shows notice about migrating to v2 to existing users
3. Error message to existing v1.1 API users while tweet post.

### Changelog Entry

> Added - Support for Twitter API v2.
> Changed - Update plugin settings and guidelines to set up a Twitter app.

### Credits

Props @iamdharmesh, @qasumitbagthariya, @ravinderk, @jeffpaul.

## Noticeable changes
- Add PHP dependency to `7.4` in composer.json

### Checklist:
